deeprank/features/Edesolv.py

```diff
@@ -0,0 +1,411 @@
+
+            # Calculate complex sasa
+            try:
+                atom.cpx_sasa = struct[0][atom.chainID][atom.resid][atom.name].sasa
+            except KeyError: #Handling alternative residues error
+                print('Alternative residue found at:', key)
+                brk_flag = True
+                pass 
+            
+            # Calculate unbound (translated) sasa
+            try:
+                atom.free_sasa = free_struct[0][atom.chainID][atom.resid][atom.name].sasa
+            except KeyError: #Handling alternative residues error
+                print('Alternative residue found at:', key)
+                brk_flag = True
+                pass 
+            
+            if not brk_flag:
+                
+                # check if the atom is a sulfide involved in a disulfide bond
+                for cys in disulfides_cys:
+                    if cys == (atom.chainID, atom.resid):
+                        disulf_bond = True
+                        
+                # Assign solvation parameters
+                Assign_solv_param(atom, disulf_bond)
+                
+                #Calculate atom complex edesolv, free edesolv and final edesolv (cpx - free)
+                atom.esolcpx = atom.cpx_sasa * atom.solv
+                atom.esolfree = atom.free_sasa * atom.solv
+                atom.edesolv = atom.esolcpx - atom.esolfree
+                esolcpx += atom.esolcpx
+                esolfree += atom.esolfree
+                
+                # xyz-val
+                # { (0|1,x,y,z) : [val] }
+                chain = [{chainA:0,chainB:1}[key[1]]]
+                k = tuple(chain + coords)
+
+                #Human readable
+                self.edesolv_data[(key[1], key[3], key[2], key[4])] = [atom.edesolv]
+                #XYZ data
+                self.edesolv_data_xyz[k] = [atom.edesolv]
+
+        self.feature_data['Edesolv'] = self.edesolv_data
+        self.feature_data_xyz['Edesolv'] = self.edesolv_data_xyz
+
+##################################################################
+
+##################################################################
+
+def __compute_feature__(pdb_data, featgrp, featgrp_raw):
+    """Main function called in deeprank for the feature calculations.
+
+    Args:
+        pdb_data (list(bytes)): pdb information
+        featgrp (str): name of the group where to save xyz-val data
+        featgrp_raw (str): name of the group where to save human readable data
+    """
+
+    edesolv_feat = Edesolv(pdb_data)
+    edesolv_feat.get_feature()
+
+    # export in the hdf5 file
+    edesolv_feat.export_dataxyz_hdf5(featgrp)
+    edesolv_feat.export_data_hdf5(featgrp_raw)
+
+    # close
+    edesolv_feat.db._close()        
+        
+class Atom():
+
+    def __init__(self,atom_specs):
+        """Atom class necessary to calculate atomic-level surface area,
+        solvation parameters and desolvation energy.
+        
+        Args:
+            atom_specs("""
```

Comment:
please complete the docsting and format it nicely :-)

Reply:
Fixed in 33d9764